frontend/tests/features/homepage.test.ts

```diff
@@ -1,15 +1,24 @@
 import { goToPage } from "tests/utils/helpers";
 import { getTestID } from "../utils/selectors";
 
+const TIMEOUT_MS = 3 * 1000;
+
 describe("Homepage", () => {
   it("renders the expected elements", async () => {
     await goToPage();
 
     await expect(page).toHaveSelector(getTestID("collections-header"));
     await expect(page).toHaveSelector(getTestID("logo"));
     await expect(page).toHaveSelector(getTestID("collection-link"));
-    await expect(page).not.toHaveSelector(getTestID("visibility-tag"));
-    await page.click(getTestID("collection-link"));
-    await expect(page.url()).toContain("collections");
+    await expect(page).not.toHaveSelector(getTestID("visibility-tag"), {
+      timeout: TIMEOUT_MS,
+    });
+
+    await Promise.all([
+      page.waitForNavigation(),
```

Comment:
we need to wait for navigation before asserting the URL. Otherwise this test will be flakey!